sharding-proxy/sharding-proxy-backend/src/main/java/org/apache/shardingsphere/shardingproxy/backend/text/TextProtocolBackendHandlerFactory.java

```diff
@@ -60,8 +66,9 @@
      * @return instance of text protocol backend handler
      */
     public static TextProtocolBackendHandler newInstance(final DatabaseType databaseType, final String sql, final BackendConnection backendConnection) {
-        if (sql.toUpperCase().startsWith(ShardingCTLBackendHandlerFactory.SCTL)) {
-            return ShardingCTLBackendHandlerFactory.newInstance(sql, backendConnection);
+        final String trimCommentSQL = trimSqlComment(sql);
+        if (trimCommentSQL.toUpperCase().startsWith(ShardingCTLBackendHandlerFactory.SCTL)) {
+            return ShardingCTLBackendHandlerFactory.newInstance(trimCommentSQL, backendConnection);
```

Comment:
so here should use contains method, then trim in ShardingCTLBackendHandlerFactory

Reply:
Yes, it looks clearer. but if sql is like 
```bash
select x from table where z="sctl:"
```
this situation is rare but also possible.

Reply:
I have understood it. so I suggest you add a SCTLUtils class, it should include 2 method.

- isSCTL(final string sql )
   use regx to judge it 
- trimComment

are you agree with it?

Reply:
Add a SCTLUtils class is a good idea! i will modify it. but i think use regx will get complicated, we need to consider sql comments, it is outside the control. on the other hand, "SCTL:" may exist in the conditional  statement. so my advice is remove comments first, and then use startWith method to judge it. what do you think of it?

Reply:
if we remove comment first, it may effect non-sctl SQL, our parse engine can recognize these 
hint syntax. so I think we should just remove comment of sctl

Reply:
Yes, so the trim method just for sctl statement,  will not affect other statements. i modified the code this way. what do you think of it?

Reply:
ok, I have got it, it's no problem.